languages/cpp/reference/exercise-concepts/reverse-string.md

```diff
@@ -0,0 +1,77 @@
+# Reverse String
+
+## Task
+
+Reverse a string
+
+For example:
+input: "cool"
+output: "looc"
+
+
+## Example
+
+`reverse_string.h`
+
+```cpp
+#if !defined(REVERSE_STRING_H)
+#define REVERSE_STRING_H
+
+#include <string>
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text);
+
+}
+
+#endif
+```
+
+`reverse_string.cpp`
+```cpp
+#include "reverse_string.h"
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text)
+{
+    return std::string(text.rbegin(), text.rend());
```

Comment:
Could the constructor be a concept by itself? I tend to think yes.

Reply:
I agree, that is correct observation!